lib/genki/http/controller.rb

```diff
@@ -34,7 +36,11 @@ def request
     end
 
     def params
-      Request.current.params
+      if request.media_type == 'application/json' && !@_raw_data
```

Comment:
I think make more sense put it on Route#process, once it already take care with params, also it will avoid the issue with `!@_raw_data`